### ISSUE TITLE
Error trying to connect on WebSphere

### ISSUE THREAD

**Issue by gknepper** [veithen/visualwas#5]
Hi,

I'm trying to connect VisualVM with visualwas plugin installed inside the DMGR machine (WAS 8.5.5.12 - J9 1.7).  This DMGR uses a non default SSL certificate and the soap port was configured in a non-standard port too (20260).  When I'm starting the VisualVM using Oracle Java 8 (jdk1.8.0_191) I accept the certificate and then I receive a error. "UndeclaredThrowableException: null".  When I ran with Oracle Java 11 (jdk-11.0.1) the visualwas plugin stay stuck in the message "Connecting". 

There's a way to turn on a debug to try to troubleshoot this errors ? 

Thanks one more time!

Gabriel.

**Comment by gknepper** [comment]
Tested from a windows machine with Oracle Java 8 (jdk1.8.0_191) and I got same "UndeclaredThrowableException: null" error.

**Comment by veithen** [comment]
The UndeclaredThrowableException exception should have been fixed by 
https://github.com/veithen/visualwas/commit/420312540971cff02459e60dd3a626e41ad4213c, which went into 2.1.0. Which version of the plugin are you using?

There are also known issues with Java 9 and above; see in particular https://github.com/veithen/visualwas/commit/b001447b0c96e7edfa281c6af02945352fc55a2b. The fixes have not been released yet; I first need to do a [release of a dependency](https://markmail.org/thread/24efhd5sdidxjeal). In the meantime you could try a snapshot from https://oss.sonatype.org/content/repositories/snapshots/com/github/veithen/visualwas/visualvm-plugin/.

**Comment by veithen** [comment]
Full stack trace (from #2):

```
java.lang.reflect.UndeclaredThrowableException
at com.sun.proxy.$Proxy18.getServerMBean(Unknown Source)
at com.github.veithen.visualwas.connector.impl.ConnectorImpl.getServerMBean(ConnectorImpl.java:63)
at com.github.veithen.visualwas.jmx.soap.SOAPJMXConnector.internalConnect(SOAPJMXConnector.java:157)
at com.github.veithen.visualwas.jmx.soap.SOAPJMXConnector.connect(SOAPJMXConnector.java:116)
at javax.management.remote.JMXConnectorFactory.connect(JMXConnectorFactory.java:270)
at com.github.veithen.visualwas.WebSpherePropertiesPanel$8.run(WebSpherePropertiesPanel.java:305)
Caused by: javax.management.InstanceNotFoundException: No MBean of type Perf found
at com.github.veithen.visualwas.connector.proxy.SingletonMBeanLocator$1$1.onSuccess(SingletonMBeanLocator.java:74)
at com.github.veithen.visualwas.connector.proxy.SingletonMBeanLocator$1$1.onSuccess(SingletonMBeanLocator.java:62)
at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1058)
at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:398)
at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1029)
at com.google.common.util.concurrent.AbstractFuture.addListener(AbstractFuture.java:675)
at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener(AbstractFuture.java:105)
at com.google.common.util.concurrent.Futures.addCallback(Futures.java:1033)
at com.github.veithen.visualwas.connector.proxy.SingletonMBeanLocator$1.onSuccess(SingletonMBeanLocator.java:60)
at com.github.veithen.visualwas.connector.proxy.SingletonMBeanLocator$1.onSuccess(SingletonMBeanLocator.java:50)
at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1058)
at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:398)
at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1029)
at com.google.common.util.concurrent.AbstractFuture.addListener(AbstractFuture.java:675)
at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.addListener(AbstractFuture.java:105)
at com.google.common.util.concurrent.Futures.addCallback(Futures.java:1033)
at com.github.veithen.visualwas.connector.proxy.SingletonMBeanLocator.locateMBean(SingletonMBeanLocator.java:48)
at com.github.veithen.visualwas.client.pmi.ConfigsLoaderInterceptor.produceValue(ConfigsLoaderInterceptor.java:40)
at com.github.veithen.visualwas.connector.feature.ContextPopulatingInterceptor.invoke(ContextPopulatingInterceptor.java:47)
at com.github.veithen.visualwas.connector.feature.ContextPopulatingInterceptor.invoke(ContextPopulatingInterceptor.java:32)
at com.github.veithen.visualwas.connector.impl.InterceptorHandler.invoke(InterceptorHandler.java:40)
at com.github.veithen.visualwas.connector.mapped.InvocationInterceptor.invoke(InvocationInterceptor.java:57)
at com.github.veithen.visualwas.connector.mapped.InvocationInterceptor.invoke(InvocationInterceptor.java:30)
at com.github.veithen.visualwas.connector.impl.InterceptorHandler.invoke(InterceptorHandler.java:40)
at com.github.veithen.visualwas.connector.impl.AdminServiceFactory$1.invoke(AdminServiceFactory.java:51)
at com.github.veithen.visualwas.framework.proxy.InvocationStyle$1.invoke(InvocationStyle.java:43)
at com.github.veithen.visualwas.framework.proxy.InvocationHandlerDelegate.invoke(InvocationHandlerDelegate.java:34)
at com.github.veithen.visualwas.framework.proxy.ProxyInvocationHandler.internalInvoke(ProxyInvocationHandler.java:57)
at com.github.veithen.visualwas.framework.proxy.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:52)
... 6 more
```

It looks like the code doesn't correctly handle the situation where PMI is disabled on the server.

**Comment by gknepper** [comment]
> The UndeclaredThrowableException exception should have been fixed by
> [4203125](https://github.com/veithen/visualwas/commit/420312540971cff02459e60dd3a626e41ad4213c), which went into 2.1.0. Which version of the plugin are you using?
> 
> There are also known issues with Java 9 and above; see in particular [b001447](https://github.com/veithen/visualwas/commit/b001447b0c96e7edfa281c6af02945352fc55a2b). The fixes have not been released yet; I first need to do a [release of a dependency](https://markmail.org/thread/24efhd5sdidxjeal). In the meantime you could try a snapshot from https://oss.sonatype.org/content/repositories/snapshots/com/github/veithen/visualwas/visualvm-plugin/.

Hi, I'm using this VisualWAS-2.1.0.nbm ( I think it's the last) with com.github.veithen.visualwas.mxbeans_2.1.0.jar. 

I'll test this new version and I'll post my findings too.

**Comment by gknepper** [comment]
Hi,

I tested visualvm-plugin-2.1.1-20181030.234149-2.nbm upgrading the previous VisualWAS-2.1.0.nbm in a windows machine and it's working nicely.

Thanks,

Gabriel.

**Comment by veithen** [comment]
Released 2.2.0 which should fix all the issues described here.
